secure/handler/authorizationHandler.go

```diff
@@ -106,8 +110,11 @@ func (a AuthorizationHandler) Decorate(delegate http.Handler) http.Handler {
 		ctx = context.WithValue(ctx, "method", request.Method)
```

Comment:
This kind of code is handled by go-kit's `transporthttp` package.  Leverage that if you can.

Also, don't use string context keys.  Use a custom type.  If you need to add lots of context variables, it's more efficient to just use a `struct` under a single key:

```
type Sat struct {
    Method string
    Path string
    ClientID string
}

// don't be afraid to pass `sat` by value.  it's efficient and ensures immutability.
func NewContextSat(ctx context.Context, sat Sat) context {
    return context.WithValue(contextKey{}, sat)
}
```